badge_server/main.py

```diff
@@ -75,519 +88,235 @@ def _get_pair_status_for_packages(pkg_sets):
-    color = badge_utils.STATUS_COLOR_MAPPING[status]
-
-    details_link = url_prefix + flask.url_for('one_badge_target',
-                                              package=package_name)
-
-    # Include the check timestamp for github head
-    if is_github and timestamp:
-        badge_name = '{} {}'.format(badge_name, timestamp)
+        {
+            'py2': { 'status': 'SUCCESS', 'details': {} },
+            'py3': { 'status': 'SUCCESS', 'details': {} },
+        }
+    """
+    result_dict = badge_utils._build_default_result(status='SUCCESS')
+    pair_mapping = badge_utils.store.get_pairwise_compatibility_for_package(
+        package_name)
+    for pair, compatibility_results in pair_mapping.items():
+        _, other_package = pair
+        if package_name == other_package.install_name:
+            other_package, _ = pair
+        unsupported_package_mapping = configs.PKG_PY_VERSION_NOT_SUPPORTED
+
+        for res in compatibility_results:
+            version = res.python_major_version            # eg. '2', '3'
+            pyver = badge_utils.PY_VER_MAPPING[version]   # eg. 'py2', 'py3'
+
+            if result_dict[pyver]['details'] is None:
+                result_dict[pyver]['details'] = {}
+
+            # Only look at a check failure status
+            # Ignore the unsupported and non self compatible packages
+            if res.status.value == 'SUCCESS':
+                continue
+
+            # Not all packages are supported in both Python 2 and Python 3. If
+            # `other_package` is not supported in the Python version being
+            # checked then skip the result.
+            unsupported_packages = unsupported_package_mapping.get(version)
+            other_package_name = other_package.install_name
+            if other_package_name in unsupported_packages:
+                continue
+
+            # If `other_package` is not self compatible (meaning that it has a
+            # conflict within it's own dependencies) then skip the result since
+            # the `other_package` will be incompatible with all other packages.
+            self_compat_res = _get_self_compatibility_dict(other_package_name)
+            if self_compat_res[pyver]['status'] != 'SUCCESS':
+                continue
+
+            details = res.details or badge_utils.EMPTY_DETAILS
+            result_dict[pyver]['status'] = res.status.value
+            result_dict[pyver]['details'][other_package_name] = details
+
+    return result_dict
+
+
+def _get_dependency_dict(package_name) -> dict:
+    """Return the dict which contains the self outdated dependencies
+    status and details.
+
+    Args:
+        package_name: the name of the package to get outdated dependencies for.
+
+    Returns:
+        A dict containing the outdated dependency status and details for any
+        outdated dependencies. The dict will be formatted like the following:
+
+        {
+            'status': 'UP_TO_DATE', 'details': {}
```

Comment:
You should show the case where there are actually outdated dependencies.

Reply:
done